src/main/resources/templates/fragments/common.html

```diff
@@ -83,6 +83,11 @@
   <script th:inline="javascript">
 
     function UpdatePosthogConsent(){
+      if(typeof(posthog) == "undefined") {
+        console.log("Posthog not initialised");
+        return;
+      }
```

Comment:
This check should be above the function. If posthog is not defined, you are asking users to consent to non existing analytics.